communications/signals.py

```diff
@@ -0,0 +1,70 @@
+from django.db.models.signals import post_save, post_migrate
+from django.dispatch import receiver
+from django.conf import settings
+from .models import UserNotificationPreference, NotificationType, UserNotificationTypePreference
+
+
+@receiver(post_save, sender=settings.AUTH_USER_MODEL)
+def create_user_notification_preferences(sender, instance, created, **kwargs):
+    """
+    Create default notification preferences and type preferences when a new user is created.
+    """
+    if created:
+        preference = UserNotificationPreference.objects.create(user=instance)
+        
+        notification_types = NotificationType.objects.filter(is_active=True)
+        for notification_type in notification_types:
+            UserNotificationTypePreference.objects.create(
+                user_preference=preference,
+                notification_type=notification_type,
+                frequency=notification_type.default_frequency
+            )
```

Comment:
When creating UserNotificationTypePreference for each type, check that duplicates are not accidentally created on repeated user creation (see above). Use get_or_create here as well for safety.